### PR TITLE
Mpifileutils

### PR DESCRIPTION
**Pull request by adammoody**
This is using some old style spack.  I realize this needs some updating, so I'm happy for any help with that.  Thanks!

### CODE REVIEW

**Comment by adamjstewart** [comment]
You basically have everything correct already if you replace the current logic with the commented out code. Let me know if you have any specific questions on how to do something!

File: var/spack/repos/builtin/packages/mpifileutils/package.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        description="Enable optimizations and features for Lustre")
+
+    # install experimental tools
+    # (coming with v0.7)
+    #variant('experimental', default=False,
+    #   description="Install experimental tools")
+
+    def configure_args(self):
+        args = [
+            "--prefix=" + prefix
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpifileutils/package.py]: `--prefix` is already added by `AutotoolsPackage` so you can remove this line.

File: var/spack/repos/builtin/packages/mpifileutils/package.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        ]
+
+        if '+lustre' in self.spec:
+            args.append('--enable-lustre')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpifileutils/package.py]: You should also explicitly _disable_ it in an else clause: we don't want non-deterministic packages that build things if they are found and don't if they aren't.

File: var/spack/repos/builtin/packages/mpifileutils/package.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        return args
+
+    def install(self, spec, prefix):
+        if '+xattr' in spec:
+            make('CFLAGS=-DDCOPY_USE_XATTRS')
+        else:
+            make()
+
+        make("install")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpifileutils/package.py]: `AutotoolsPackage` comes with 3 phases: configure, build, and install (well, technically it comes with autoreconf too). The default for `build` is `make()`. The default for `install` is `make('install')`. With your current logic, this means you would be running `make` 3 times, not 2. To get what you want, I recommend dropping this `install` method altogether and adding:
```python
@property
def build_targets(self):
    targets = []
    if '+xattr' in self.spec:
        targets.append('CFLAGS=-DDCOPY_USE_XATTRS')
    return targets
```

File: var/spack/repos/builtin/packages/mpifileutils/package.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    version('0.6', '620bcc4966907481f1b1a965b28fc9bf')
+
+    depends_on('mpi')
+    depends_on('libcircle')
+    depends_on('lwgrp')
+    depends_on('dtcmp')
+    depends_on('libarchive')
+
+    # enable optimizations / features for Lustre
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mpifileutils/package.py]: Is this comment wrong? I don't think you really need a comment anyway. That's what the description is for.

**Comment by adammoody** [comment]
@adamjstewart Thanks for your help!

**Pull request merged**